src/bacnet/basic/object/device.c

```diff
@@ -76,12 +76,21 @@
 #include "bacnet/basic/object/bacfile.h"
 #endif /* defined(BACFILE) */
 
+#ifndef __ZEPHYR__
```

Comment:
@Michail-Antropov Forcing `USE_OBJECT_TABLE_ARRAY` needs be done from the top-level build files (e.g., CMakeLists.txt) so that it is consistent for everything built.  Please relocate.